hbase-zookeeper/src/main/java/org/apache/hadoop/hbase/zookeeper/ZKWatcher.java

```diff
@@ -692,12 +692,12 @@ public void interruptedExceptionNoThrow(InterruptedException ie, boolean throwLa
    */
   @Override
   public void close() {
-    zkEventProcessor.shutdownNow();
     try {
       recoverableZooKeeper.close();
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
     }
```

Comment:
oh wait, can we put `shutdownNow()` to `finally` block? Even without that it should be executed but just in case in future, more Exception handling might be lined up or if ZK client close throws some runtime exception, at least we will perform shutdown of threadpool in finally.

Reply:
good idea, thanks!